ticker/src/main/java/com/robinhood/ticker/LevenshteinUtils.java

```diff
@@ -12,26 +30,124 @@
+                Arrays.fill(remainingActions, ACTION_INSERT);
+                columnActions.add(remainingActions);
+                totalNumActions += remainingActions.length;
+                break;
+            } else if (reachedEndOfTarget) {
+                final int[] remainingActions = new int[source.length - sourceIndex];
+                Arrays.fill(remainingActions, ACTION_DELETE);
+                columnActions.add(remainingActions);
+                totalNumActions += remainingActions.length;
+                break;
+            }
+
+            final boolean containsSourceChar = supportedCharacters.contains(source[sourceIndex]);
+            final boolean containsTargetChar = supportedCharacters.contains(target[targetIndex]);
+
+            if (containsSourceChar && containsTargetChar) {
+                // We reached a segment that we can perform animations on
+                final int sourceEndIndex =
+                        findNextUnsupportedChar(source, sourceIndex + 1, supportedCharacters);
+                final int targetEndIndex =
+                        findNextUnsupportedChar(target, targetIndex + 1, supportedCharacters);
+                final int[] columnActionsForSegment = computeColumnActionsForSegment(
+                        source, target, sourceIndex, sourceEndIndex, targetIndex, targetEndIndex
+                );
+                columnActions.add(columnActionsForSegment);
+                totalNumActions += columnActionsForSegment.length;
+                sourceIndex = sourceEndIndex;
+                targetIndex = targetEndIndex;
+            } else if (containsSourceChar) {
+                // We are animating in a target character that isn't supported
+                columnActions.add(new int[] { ACTION_INSERT });
+                totalNumActions++;
+                targetIndex++;
+            } else if (containsTargetChar) {
+                // We are animating out a source character that isn't supported
+                columnActions.add(new int[] { ACTION_DELETE });
+                totalNumActions++;
+                sourceIndex++;
+            } else {
+                // Both characters are not supported, perform default animation to replace
+                columnActions.add(new int[] { ACTION_SAME });
+                totalNumActions++;
+                sourceIndex++;
+                targetIndex++;
+            }
+        }
+
+        // Concat all of the actions into one array
+        final int[] result = new int[totalNumActions];
+        int index = 0;
+        for (int[] actions : columnActions) {
+            System.arraycopy(actions, 0, result, index, actions.length);
+            index += actions.length;
+        }
+        return result;
+    }
+
+    private static int findNextUnsupportedChar(char[] chars, int startIndex,
+            Set<Character> supportedCharacters) {
+        while (startIndex < chars.length) {
```

Comment:
I think this would read better as a for loop
```
for (int i = startIndex; i < chars.length; i++) {
  if (!supportedChars.contains(chars[i]) {
    return i;
  }
}
```